File: src/rocketsmith/gui/layout.py
"""Canonical project directory layout.

Tools use these as default output subdirectories.  Every path is relative
to the project root.  Users can override individual tool outputs with
an explicit ``out_path`` parameter.
"""

OPENROCKET_DIR = "openrocket"
FLIGHTS_DIR = "openrocket/flights"

# GUI directory — holds the frontend bundle, data snapshot, and derived data.
GUI_DIR = "gui"
GUI_ASSETS_DIR = "gui/assets"
TREE_FILE = "gui/component_tree.json"
ASSEMBLY_FILE = "gui/assembly.json"
GUI_PID_FILE = "gui/.gui.pid"
GUI_MAIN_JS = "gui/main.js"
GUI_DATA_JS = "gui/data.js"

# Per-part JSON metadata directory.
PARTS_DIR = "gui/parts"

# Preview / visualization assets (under gui/assets/).
STL_DIR = "gui/assets/stl"
PNG_DIR = "gui/assets/png"
GIF_DIR = "gui/assets/gif"
TXT_DIR = "gui/assets/txt"
PROGRESS_DIR = "gui/progress"

# Logs directory for agentic session logs.
LOGS_DIR = "gui/logs"

# CAD directories — scripts and geometry.
CADSMITH_DIR = "cadsmith"
CADSMITH_SOURCE_DIR = "cadsmith/source"
STEP_DIR = "cadsmith/step"

# Slicer directories — configs and output.
PRUSASLICER_DIR = "prusaslicer"
PRUSASLICER_CONFIG_DIR = "prusaslicer/configs"
GCODE_DIR = "prusaslicer/gcode"

File: src/rocketsmith/gui/web/src/layout.ts
/** Canonical project directory layout — mirrors gui/layout.py. */

export const OPENROCKET_DIR = "openrocket";

// GUI directory — holds the frontend bundle, data snapshot, and derived data.
export const GUI_DIR = "gui";
export const GUI_ASSETS_DIR = "gui/assets";
export const TREE_FILE = "gui/component_tree.json";
export const ASSEMBLY_FILE = "gui/assembly.json";

// Per-part JSON metadata directory.
export const PARTS_DIR = "gui/parts";

// Preview / visualization assets (under gui/assets/).
export const STL_DIR = "gui/assets/stl";
export const PNG_DIR = "gui/assets/png";
export const GIF_DIR = "gui/assets/gif";
export const TXT_DIR = "gui/assets/txt";
export const PROGRESS_DIR = "gui/progress";

// CAD directories — scripts and geometry.
export const CADSMITH_DIR = "cadsmith";
export const CADSMITH_SOURCE_DIR = "cadsmith/source";
export const STEP_DIR = "cadsmith/step";

// Slicer directories — configs and output.
export const PRUSASLICER_DIR = "prusaslicer";
export const PRUSASLICER_CONFIG_DIR = "prusaslicer/config";
export const GCODE_DIR = "prusaslicer/gcode";
